src/js/index.js

```diff
@@ -0,0 +1,109 @@
+const playerFactory = (name, symbol) => {
+  const getName = () => name;
+  const getSymbol = () => symbol;
+  return { getName, getSymbol };
+};
+
+const gameFactory = () => {
+  let boardArray = [['', '', ''], ['', '', ''], ['', '', '']];
+  const player = [];
+  let turn = 0;
+  let status = 'pending';
+
+  const renderBoard = () => {
+    boardArray.forEach((row, rowIndex) => {
+      row.forEach((cellValue, columnIndex) => {
+        document.querySelector('#board')
+          .querySelector(`.board-row:nth-child(${rowIndex + 1}`)
+          .querySelector(`.board-cell:nth-child(${columnIndex + 1}`)
+          .innerHTML = cellValue;
+      });
+    });
+  };
+
+  const sendMessage = (msg) => {
+    document.getElementById('message').innerHTML = `<p>${msg}</p>`;
+  };
+
+  const isWinner = () => {
+    let finished = false;
+    const h = ['00-01-02', '10-11-12', '20-21-22']; // horizontalWinningCombinations
+    const v = ['00-10-20', '01-11-21', '02-12-22']; // verticalWinningCombinations
+    const d = ['00-11-22', '20-11-02']; // diagonalWinningCombinations
+    [...h, ...v, ...d].forEach((move) => {
+      const line = move.split('-');
+      if (boardArray[line[0][0]][line[0][1]] === boardArray[line[1][0]][line[1][1]]
+        && boardArray[line[1][0]][line[1][1]] === boardArray[line[2][0]][line[2][1]]
+        && boardArray[line[0][0]][line[0][1]] !== '') {
+        finished = true;
+      }
+    });
+    return finished;
+  };
+
+  const isDraw = () => !boardArray.some(line => line.includes(''));
+
+  const finish = type => {
+    status = 'pending';
+    if (type === 'draw') {
+      sendMessage('Game finished, its a draw!');
+    } else {
+      sendMessage(`Game finished, ${player[turn].getName()} won.`);
+    }
+  };
+
+  const play = move => {
+    if (status !== 'running') sendMessage('Start the game first!');
+
+    boardArray[move[0]][move[1]] = player[turn].getSymbol();
+    renderBoard();
```

Comment:
Here (line 58) you just collect the player's symbol and fill the board with. But you haven't controlled the fact that the other player can also click on a square that is already clicked, and this will change the symbol already there, and this can influence the game.

I count on you to implement this control 🎅